From e529a183caa102a626e8df40ebda8e23f9d1b63c Mon Sep 17 00:00:00 2001
From: Guan Jing <guanjing6@huawei.com>
Date: Tue, 10 May 2022 23:04:55 +0800
Subject: [PATCH] sched: Introduce qos smt expeller for co-location
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: e529a183caa102a626e8df40ebda8e23f9d1b63c
Modified-by-SEL: No


hulk inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I52611
CVE: NA

--------------------------------

We introduce the qos smt expeller, which lets
online tasks to expel offline tasks on the smt sibling cpus,
and exclusively occupy CPU resources.In this way we are
able to improve QOS of online tasks in co-location.

Signed-off-by: Guan Jing <guanjing6@huawei.com>
Reviewed-by: Chen Hui <judy.chenhui@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 init/Kconfig | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/init/Kconfig b/init/Kconfig
index 895e0ef85f73..27c5ed16fef1 100644
--- a/init/Kconfig
+++ b/init/Kconfig
@@ -966,6 +966,15 @@ config QOS_SCHED
 
     default n
 
+config QOS_SCHED_SMT_EXPELLER
+	bool "Qos smt expeller"
+	depends on SCHED_SMT
+	depends on QOS_SCHED
+	default n
+	help
+	  This feature enable online tasks to expel offline tasks
+	  on the smt sibling cpus, and exclusively occupy CPU resources.
+
 config FAIR_GROUP_SCHED
 	bool "Group scheduling for SCHED_OTHER"
 	depends on CGROUP_SCHED
-- 
2.34.1

